phoenix-core/src/test/java/org/apache/phoenix/util/TestUtil.java

```diff
@@ -934,7 +934,30 @@ public IndexStateCheck(PIndexState indexState, Long indexDisableTimestamp, Boole
     		this.success = success;
     	}
     }
-    
+
+    public static void waitForIndexState(Connection conn, String fullIndexName, PIndexState expectedIndexState) throws InterruptedException, SQLException {
+        int maxTries = 60, nTries = 0;
+        do {
+            String schema = SchemaUtil.getSchemaNameFromFullName(fullIndexName);
+            String index = SchemaUtil.getTableNameFromFullName(fullIndexName);
+            Thread.sleep(1000); // sleep 1 sec
+            String query = "SELECT " + PhoenixDatabaseMetaData.INDEX_STATE + " FROM " +
+                    PhoenixDatabaseMetaData.SYSTEM_CATALOG_NAME + " WHERE (" + PhoenixDatabaseMetaData.TABLE_SCHEM + "," + PhoenixDatabaseMetaData.TABLE_NAME
+                    + ") = (" + "'" + schema + "','" + index + "') "
+                    + "AND " + PhoenixDatabaseMetaData.COLUMN_FAMILY + " IS NULL AND " + PhoenixDatabaseMetaData.COLUMN_NAME + " IS NULL";
+            ResultSet rs = conn.createStatement().executeQuery(query);
+            PIndexState actualIndexState = null;
+            if (rs.next()) {
+                actualIndexState = PIndexState.fromSerializedValue(rs.getString(1));
+                boolean matchesExpected = (actualIndexState == expectedIndexState);
+                if (matchesExpected) {
+                    return;
+                }
+            }
+        } while (++nTries < maxTries);
```

Comment:
Do you wanna utilize exponential backoff? Maybe worth looking at hbase's retryConfig utility.

Reply:
I am not sure if exponential backoff is necessary here. It doesn't take a lot of time to wait till indexes are build in the tests.

Reply:
I see there are 60 maxRetries. This might end up using resources unnecessarily. It's better to utilize an existing retry framework rather than reinventing the wheel. RetryConfig gives an option to specify the policy of the retry mechanism. What do you suggest @gjacoby126?

Reply:
@swaroopak - as we talked about offline, RetryCounter has InterfaceAudience.Private, which is HBase's way of telling us that external projects shouldn't depend on it because it's an internal implementation detail that can go away or change interface at any time. 

Would be nice to have something similar in Phoenix though, so perhaps we should fork it (as part of a different JIRA).